arch/arm64/src/common/arm64_arch_timer.c

```diff
@@ -286,9 +286,8 @@ static int arm64_tick_start(struct oneshot_lowerhalf_s *lower,
 
   priv->running = this_cpu();
 
-  next_cycle =
-    arm64_arch_timer_count() / priv->cycle_per_tick * priv->cycle_per_tick +
-    ticks * priv->cycle_per_tick;
+  next_cycle = (clock_systime_ticks() + ticks) *
```

Comment:
This puts bsck the timer drift; there is an unknown delay between tick interrupt taken and this one called, which caises the drift.

The current code fixes the drift issue by flooring the current time to an even multiple of tick time; thus also the next_cycle is at even multiple of tick time, and each cycle is exactly of correct length. Also querying the current tick always then returns the same number during the tick period.